### ISSUE TITLE
Safari 8 completely broken?

### ISSUE THREAD

**Issue by milkisevil** [google/shaka-player#169]
I'm testing in Safari 8.0.8 (osx 10.10.5) and none of the samples on http://shaka-player-demo.appspot.com/ are working (except GPAC which has seeking problems). Also tested in Chrome where the samples work as normal. Is Shaka Player completely broken in Safari 8?

**Comment by joeyparrish** [comment]
Looks like you aren't the only one having issues with this.  There's a [thread on the mailing list this morning](https://groups.google.com/d/msgid/shaka-player-users/2a4f85c5-e979-43f3-8d64-6901f2ad151c%40googlegroups.com) about Safari 8 issues.

We don't test Safari 8 explicitly, but our aim is to [support the browsers that support the required standards](http://shaka-player-demo.appspot.com/docs/tutorial-porting.html), such as MSE.

If this worked before, you can test in older Shaka Player releases using appspot.  For example, you can visit http://v1-3-0.shaka-player-demo.appspot.com/ to test v1.3.0.  Please test on older releases (1.4.0, 1.3.2, 1.3.1, 1.30, 1.2.4, ...) and let us know if you find a version that works on Safari.  We can debug from there.

**Comment by joeyparrish** [comment]
Hi @milkisevil, were you able to test older versions of Shaka Player on Safari 8?  Do we have a regression?

**Comment by milkisevil** [comment]
Sorry for the delay. No luck with Safari 8.0.8 across any of the Shaka versions. The `GPAC/SegmentList` video is the most likely to begin playback across versions, but will always throw an error when you attempt to seek. The `"Car" (YT DASH test)` doesn't start on any Shaka player version.

I've also tested with Safari 9 (11601.1.39.0.2) via [Browserstack](www.browserstack.com) and getting better results, but still quite erratic. `"Car" (YT DASH test)` will start playback but when seeking, shaka will throw the following error:

```
Multiple buffered ranges detected: (3)
"Either the content has gaps in it,"
"the content's segments are not aligned across bitrates,"
"or the browser has evicted the middle of the buffer."
gshaka-player-demo.appspot.com:74:177
```

**Comment by joeyparrish** [comment]
Full Safari support is a goal of v2.0.

**Comment by jonobr1** [comment]
I'm not sure if this is related, but I think it is. `v1.6.3` of Shaka Player throws `DOM Exception 18` when uploading video as a `texture2d` to WebGL.

I looked at the [`REDESIGN.md`](https://github.com/google/shaka-player/blob/preview/docs/design/REDESIGN.md) and it says sometime in January 2016 `v2.0.+` will be released. Is there any update on this?

**Comment by joeyparrish** [comment]
Shaka doesn't use WebGL.  Are you sure that's related?

The schedule in the design doc needs to be updated.  If you check our [project milestones](https://github.com/google/shaka-player/milestones), our current target for a v2 beta is the end of February, with a full release by the end of March.

You can see progress to-date in the 'preview' branch.  We've been working our way upward, and we're getting very close to the top of the stack.

**Comment by jonobr1** [comment]
oh awesome, very handy. Thanks @joeyparrish!

I know Shaka doesn't _use_ WebGL, but I noticed that when I updated to `v1.6.3` I started getting these errors in Safari. I still need to do some research, but it's related in the sense that Shaka Player and Safari are involved. Sorry for the noise!

**Comment by joeyparrish** [comment]
@jonobr1 Can you tell me where in the Shaka codebase this exception originates?  Is there a stack trace?  Are there any console logs from Shaka before this happens?

All of our automated tests for v2 so far are passing in Safari 8, but if we can understand what the issue is in v1, I'd love to publish a fix before v2 is out.

**Comment by jonobr1** [comment]
Certainly! I'm making viewable examples, so that should be much clearer than bits and pieces. I'll share them once I've finished.

**Comment by jonobr1** [comment]
![screen shot 2016-02-11 at 5 46 00 pm](https://cloud.githubusercontent.com/assets/220033/12996733/57c2d2dc-d0e7-11e5-8ad6-a77d90774080.png)

https://dl.dropboxusercontent.com/u/1595444/shaka-player/stream-texture.html

Now that I've done some more testing it seems like my previous version of the Shaka Player also has the `DOM Exception 18` error. So, I think you're right — it has to do more with WebGL than with Shaka. I'm gonna post this over on the Three.js issues.

**Comment by joeyparrish** [comment]
@jonobr1 so it seems now that we've confirmed that your WebGL issues are unrelated to Shaka.  Do you have issues on Safari 8 in the latest version of the [standard demo app](http://shaka-player-demo.appspot.com/)?

**Comment by joeyparrish** [comment]
Update: I got my hands on Safari 8 and tried the latest release.  I found that Safari's CORS implementation is subtly different from Chrome's.  A CORS preflight request from Safari 8 includes the "Origin" header in its list of cross-origin request headers, while Chrome does not.

I've updated the CORS settings on the server that hosts some of our test assets (Sintel, for example) to explicitly list "Origin" in its list of permitted cross-origin request headers.  This seems to fix download of those assets on Safari without updates to Shaka.

Playback still seems to be broken on Safari 8, but there are no error messages from the browser.  I'm not sure what the issue is with Safari 8.

In **Safari 9**, with the server update, **playback of Sintel is now working.**

Some of the other test assets don't work on Safari, but those same assets also do not work on IE11.  (Both produce "multiple buffered range" errors from Shaka because they are seen by the browser as having discontinuous segments.)  My *theory* is that these test assets (the car clips, for example) are not 100% correct, but work on Chrome because Chrome's parsers and decoders are more flexible/forgiving than some others.
